app/assets/javascripts/utils/statistic_update_info_utils.js

```diff
@@ -19,11 +19,12 @@ const getLastUpdateMessage = (course) => {
   let nextUpdateMessage = '';
   let isNextUpdateAfter = false;
   const lastUpdateMoment = lastSuccessfulUpdateMoment(course.flags.update_logs);
+  const updatesEndMoment = toDate(course.update_until);
   if (lastUpdateMoment) {
     const averageDelay = course.updates.average_delay ?? 0;
     lastUpdateMessage = `${I18n.t('metrics.last_update')}: ${formatDistanceToNow(lastUpdateMoment, { addSuffix: true })}.`;
     const nextUpdateExpectedTime = addSeconds(lastUpdateMoment, averageDelay);
-    isNextUpdateAfter = isAfter(nextUpdateExpectedTime, new Date());
+    isNextUpdateAfter = (isAfter(nextUpdateExpectedTime, new Date()) && isAfter(updatesEndMoment, new Date()));
```

Comment:
This works as expected, but it's not going to be obvious to future coders what the purpose of this line is, and the logic is duplicated in the other method.

Can the check against the end-of-updates date be extracted to another method so there's less duplication and there's a method name that makes the purpose more clear?

Reply:
sure, i have extracted it to another method.